From b8f9919f000e642fc8970d0a42c1bdadf5a9d720 Mon Sep 17 00:00:00 2001
From: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Date: Mon, 10 Aug 2020 08:30:34 +0200
Subject: [PATCH] sd-daemon.c: Fix build with newer glibc and musl libc

Reported by Khem Raj on meta-oe list

TOPDIR/build/tmp/work/armv7vet2hf-neon-yoe-linux-musleabi/dlt-daemon/2.18.5-r0/recipe-sysroot/usr/include/sys/fcntl.h:1:2: error: redirecting incorrect #include <sys/fcntl.h> to <fcntl.h> [-Werror,-W#warnings]
 ^
1 error generated.

Signed-off-by: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Signed-off-by: Gianfranco Costamagna <locutusofborg@debian.org>
---
 systemd/3rdparty/sd-daemon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/systemd/3rdparty/sd-daemon.c b/systemd/3rdparty/sd-daemon.c
index 88936d2..de3f985 100644
--- a/systemd/3rdparty/sd-daemon.c
+++ b/systemd/3rdparty/sd-daemon.c
@@ -35,7 +35,7 @@
 #ifdef __BIONIC__
 #include <linux/fcntl.h>
 #else
-#include <sys/fcntl.h>
+#include <fcntl.h>
 #endif
 #include <netinet/in.h>
 #include <stdlib.h>
